src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -679,9 +679,14 @@ namespace cryptonote
     r = m_mempool.init(max_txpool_weight, m_nettype == FAKECHAIN);
     CHECK_AND_ASSERT_MES(r, false, "Failed to initialize memory pool");
 
+    session_start_height = get_current_blockchain_height();
+    MGINFO_YELLOW("Current height of the chain : " << session_start_height);
```

Comment:
I was thinking about what happens if I start the daemon, let it sync for a while, but before it reaches top of chain, I exit and then restart.

I am pretty sure nothing really bad happens. The daemon just can remove away less transactions from the pool, because the `session_start_hight` of the first, incomplete run is of course lost, and the second start height gets used fro pool cleaning.